docs/concepts/tx-lifecycle.md

```diff
@@ -0,0 +1,231 @@
+
+If `Tx` is included in a block and full-nodes receive precommits from +2/3 of the validators, the nodes understand to execute the commit stage. `Tx` officially becomes part of the blockchain, which logs the application's transaction history.
+
+## State Changes
+
+During consensus, all validator-nodes come to agreement on not only whether `Tx` is to be included but also the precise order of the transactions in its block. However, apart from committing to this block in consensus, the ultimate goal is actually for full-nodes to commit to a new state generated by `Tx` state changes.
+In order to execute `Tx`, the following ABCI function calls are made in order to communicate to the application what state changes need to be made. While full-nodes each run everything individually, since the messages' state transitions are deterministic and the order was finalized during consensus, this process yields a single, unambiguous result.
+```
+		-----------------------		
+		| BeginBlock	      |        
+		-----------------------		
+		          |		
+			  v			
+		-----------------------		    
+		| DeliverTx(tx0)      |  
+		| DeliverTx(tx1)      |   	  
+		| DeliverTx(tx2)      |  
+		| DeliverTx(tx3)      |  
+		|	.	      |  
+		|	.	      |
+		|	.	      |
+		-----------------------		
+		          |			
+			  v			
+		-----------------------
+		| EndBlock	      |         
+		-----------------------
+		          |			
+			  v			
+		-----------------------
+		| Commit	      |         
+		-----------------------
+```
+### Block Functions
+
+There are a few other functions that the full-nodes run to commit the block.
+[`BeginBlock`](./app-anatomy.md#beginblocker-and-endblocker) is always run before `DeliverTx`, and [`EndBlock`](./app-anatomy.md#beginblocker-and-endblocker) afterward.
+
+### DeliverTx
+
+The `DeliverTx` ABCI function defined in [`baseapp`](./baseapp.md) does the bulk of the state change work: it is run for each transaction in the block in sequential order as committed to during consensus. Under the hood, `DeliverTx` is almost identical to `CheckTx` but calls the [`runTx`](./baseapp.md#runtx) function in deliver mode instead of check mode. Instead of using their `checkTxState` or `queryState`, full-nodes select a new copy, `deliverTxState`, to deliver transactions:
+
+* **Decoding:** Since `DeliverTx` is an ABCI call, `Tx` is received in the encoded `byte []` form. Nodes first unmarshal the transaction, then call `runTx` in `runTxModeDeliver`, which does everything that `CheckTx` did and also executes and writes state changes.
+* **Checks:** The full-nodes call `validateBasicMsgs` and the `AnteHandler` again. This second check happens because they may not have seen the same transactions during the Addition to Mempool stage and a malicious proposer may have included invalid transactions.
```

Comment:
Slight difference: `AnteHandler` will not check that `gas-prices > min-gas-prices`, since `min-gas-prices` is local to each node, and therefore can't be in consensus